core/include/thread.h

```diff
@@ -94,6 +94,32 @@
 #define THREAD_PRIORITY_MAIN           (THREAD_PRIORITY_MIN - (SCHED_PRIO_LEVELS/2))
 
 /**
+ * @name Optional flags for controlling a threads initial state
+ * @{
+ */
+/**
+ * @brief Set the new thread to sleeping
+ **/
+#define THREAD_CREATE_SLEEPING          (1)
+
+/**
+ * @brief Currently not implemented
+ */
+#define THREAD_AUTO_FREE                (2)
```

Comment:
Had the same in mind, but I think it makes sense to do this in a separate PR - so my plan was to merge this first and open a second one afterwards